datafusion/core/src/physical_plan/file_format/file_stream.rs

```diff
@@ -179,6 +227,14 @@ impl<F: FileOpener> FileStream<F> {
                     partition_values,
                 } => match ready!(reader.poll_next_unpin(cx)) {
                     Some(result) => {
+                        if let Some(instant) =
+                            self.file_stream_metrics.time_scanning.1.take()
+                        {
+                            self.file_stream_metrics
+                                .time_scanning
+                                .0
+                                .add_elapsed(instant);
+                        }
```

Comment:
I wonder does it matter if we update these metrics on plan failure?  As in if the stream is never ready the 'file opening time' will never get updated. Maybe that is ok 🤔